lib/dgram.js

```diff
@@ -272,6 +267,41 @@ Socket.prototype.send = function(buffer,
   if (offset + length > buffer.length)
     throw new RangeError('Offset + length beyond buffer length');
 
+  return buffer.slice(offset, offset + length);
+}
+
+Socket.prototype.send = function(buffer,
+                                 offset,
+                                 length,
+                                 port,
+                                 address,
+                                 callback) {
+  var self = this;
+
+  if (arguments.length === 5 || arguments.length === 6) {
+    buffer = sliceBuffer(buffer, offset, length);
```

Comment:
> If you're gonna send an array of buffers, I have no idea what the point of offset and length should be. It seems a rather useless feature (hence my initial proposal for a method called sendv instead). I honestly have no opinion as to what the semantics here should be, but if you can find any that make any sense at all, go for it. I don't think unix's sendmsg function actually takes these arguments when sending msg_iov.

I would really try to avoid code duplication for both cases, as I would not like to maintain both a 'send' and 'sendv' in C++ land. This is basically why I am asking this question. This is only relevant to the internal C++ API, not to the user-facing API, which would just be an array of buffers.

> The fact is though, that libuv only really seems to support sendmsg with a message object, and not a call with offset and length. The fact that Node exposes these may be legacy reasons (I don't know the history of it but can imagine that before libuv the sendmsg syscall was used differently).

This is not correct. The point where we are currently using offset and length in C++ is: https://github.com/nodejs/node/blob/master/src/udp_wrap.cc#L267-L268. That structure that goes deep down into sendmsg, and it is where the magic happens. This is faster than slicing in JS land, however I have no idea how faster it is.

IMHO, having offset and length are kind of useless anyway because of UDP. Does anybody has a lib that is using those heavily to test it?